djangoessentials/djangoessentials.py

```diff
@@ -15,3 +17,40 @@ class MyS3Storage(S3Boto3Storage):
     location = 'media/'  # S3'te files alt dir
     file_overwrite = False  # if have a same name about the file cant overwrite
     default_acl = 'public-read' # just can read permission
+
+
+
+#CustomUser Model
+class CustomUserManager(BaseUserManager):
+    def create_user(self, username, password=None, **extra_fields):
+        if not username:
+            raise ValueError('The Username field must be set')
+        user = self.model(username=username, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+    
+    def create_superuser(self, username, password=None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
+        return self.create_user(username, password, **extra_fields)
+
+
+class CustomUser(AbstractBaseUser, PermissionsMixin):
+    username = models.CharField(max_length=150, unique=True)
+    phone_number = models.CharField(max_length=15, blank=True, null=True)
+    date_of_birth = models.DateField(blank=True, null=True)
+    is_active = models.BooleanField(default=True)
+    is_staff = models.BooleanField(default=False)
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
```

Comment:
Thank you for collabration. Every part amazing. As u know we have a `TimeBasedStampModel` so you can fix the with it (with inheritance).

The other part looks like great for me!